### ISSUE TITLE
Create an InputStation Diagram for every console controller currently used in RetroPie Libretro cores

### ISSUE THREAD

**Issue by HerbFargus** [taalas/RetroPie-Artwork#2]
The following is a list of all the consoles RetroArch Emulates: Items in bold indicate that InputStation diagrams have been created and added to the RetroPie Wiki.
### RetroArch Consoles
- [**3do**](https://github.com/petrockblog/RetroPie-Setup/wiki/3do)
- [**Atari 2600**](https://github.com/petrockblog/RetroPie-Setup/wiki/Atari-2600)
- [**Atari 7800**](https://github.com/RetroPie/RetroPie-Setup/wiki/Atari-7800)
- [**Atari Jaguar**](https://github.com/RetroPie/RetroPie-Setup/wiki/Atari-Jaguar)
- [**Atari Lynx**](https://github.com/RetroPie/RetroPie-Setup/wiki/Atari-Lynx)
- Bandai WonderSwan (Color)
- [**Genesis/Megadrive**](https://github.com/petrockblog/RetroPie-Setup/wiki/Genesis-Megadrive)
  - [**Sega32x**](https://github.com/petrockblog/RetroPie-Setup/wiki/Sega-32X)
  - [**SegaCD**](https://github.com/petrockblog/RetroPie-Setup/wiki/Sega-CD)
- [**Game Gear**](https://github.com/RetroPie/RetroPie-Setup/wiki/Game-Gear)
- [**Game Boy**](https://github.com/petrockblog/RetroPie-Setup/wiki/Game-Boy)
- [**Game Boy Color**](https://github.com/petrockblog/RetroPie-Setup/wiki/Game-Boy-Color)
- [**Game Boy Advance**](https://github.com/petrockblog/RetroPie-Setup/wiki/Game-Boy-Advance)
- [**MasterSystem**](https://github.com/petrockblog/RetroPie-Setup/wiki/Master-System)
- [**Nintendo 64**](https://github.com/petrockblog/RetroPie-Setup/wiki/Nintendo-64)
- [**Nintendo DS**](https://github.com/RetroPie/RetroPie-Setup/wiki/Nintendo-DS)
- [**Nintendo Entertainment System**](https://github.com/petrockblog/RetroPie-Setup/wiki/Nintendo-Entertainment-System)
- [**Neo Geo**](https://github.com/RetroPie/RetroPie-Setup/wiki/Neo-Geo)
- [**Neo Geo Pocket**](https://github.com/RetroPie/RetroPie-Setup/wiki/Neo-Geo-Pocket)
- [**PC Engine/ TurboGrafx 16**](https://github.com/RetroPie/RetroPie-Setup/wiki/PC-Engine)
- [**PlayStation 1**](https://github.com/RetroPie/RetroPie-Setup/wiki/Playstation-1)
- [**Sega Saturn**](https://github.com/petrockblog/RetroPie-Setup/wiki/Saturn)
- [**Sega SG-1000**](https://github.com/RetroPie/RetroPie-Setup/wiki/SG-1000)
- [**Super Nintendo Entertainment System**](https://github.com/petrockblog/RetroPie-Setup/wiki/Super-Nintendo-Entertainment-System)
- [**Vectrex**](https://github.com/RetroPie/RetroPie-Setup/wiki/Vectrex)
- [**Videopac or Odyssey2**](https://github.com/RetroPie/RetroPie-Setup/wiki/VideoPac-or-Odyssey-2)
- [**Virtual Boy**](https://github.com/RetroPie/RetroPie-Setup/wiki/Virtual-Boy)
### Non RetroArch Consoles:
- [**Dreamcast**](https://github.com/RetroPie/RetroPie-Setup/wiki/Dreamcast)
- [**Intellivision**](https://github.com/RetroPie/RetroPie-Setup/wiki/Intellivision)

Note: Will add final products to new section (InputStation Controller Diagrams) of each emulator page in the [Retropie Wiki](https://github.com/petrockblog/RetroPie-Setup/wiki)
## 3do

![3dodiagram](https://cloud.githubusercontent.com/assets/10035308/8237801/0ae4d6fc-15af-11e5-803f-8ba408d48362.png)
## Atari 2600

![atari2600diagram](https://cloud.githubusercontent.com/assets/10035308/8237960/02aa13fc-15b0-11e5-92c2-311e8960883b.png)
## Atari 7800

![atari7800diagram](https://cloud.githubusercontent.com/assets/10035308/8268732/d7b4236c-174a-11e5-8386-6fa948235267.png)
## Atari Lynx

![atari_lynx](https://cloud.githubusercontent.com/assets/10035308/8176465/8d4a4c74-13b8-11e5-9956-c059ac3fc9e1.png)
## Atari Jaguar

![atarijaguardiagram](https://cloud.githubusercontent.com/assets/10035308/8268598/4a5d1868-1748-11e5-994d-e0d508d8877b.png)
## Gameboy

![gameboy](https://cloud.githubusercontent.com/assets/10035308/7334402/bd640072-eb4e-11e4-8251-d2bc3b876153.png)
## Gameboy Color

![gameboycolor](https://cloud.githubusercontent.com/assets/10035308/7334404/bd65e496-eb4e-11e4-82e6-78494534d305.png)
## Gameboy Advance

![gameboyadvance](https://cloud.githubusercontent.com/assets/10035308/7334407/bd688f52-eb4e-11e4-8899-246bad0ad139.png)
## Game Gear

![segagamegeardiagram](https://cloud.githubusercontent.com/assets/10035308/8245009/b4c87e72-15e5-11e5-8ee5-691daa4d1dd5.png)
## Mastersystem

![mastersystem](https://cloud.githubusercontent.com/assets/10035308/7334954/59794038-eb60-11e4-989d-6367c1fda8ef.png)
## Megadrive/Genesis (3 Button)

![genesis](https://cloud.githubusercontent.com/assets/10035308/7336303/aec335e0-ebb4-11e4-93b3-26037dd26ffb.png)
## Megadrive/Genesis (6 Button)

![genesis6btn](https://cloud.githubusercontent.com/assets/10035308/7336429/7e524110-ebbb-11e4-8777-05a824384d34.png)
## Megadrive/Genesis (6 Button) Arcadepad

![megadrive6btnarcadepaddiagram](https://cloud.githubusercontent.com/assets/10035308/8268483/8b1b6dae-1744-11e5-9407-df58e2a81aad.png)
## Nintendo 64

![nintendo_n64_diagram](https://cloud.githubusercontent.com/assets/10035308/7334793/6b406d8c-eb5b-11e4-8474-56a340521e71.png)
## Nintendo DS

![dsdiagram](https://cloud.githubusercontent.com/assets/10035308/8240819/99893d9a-15c2-11e5-9d52-c86db39493f6.png)
## NES

![nesdiagram](https://cloud.githubusercontent.com/assets/10035308/8245062/4f0c5b8e-15e6-11e5-9255-b920543518d6.png)
## Neo Geo

![neogeodiagram](https://cloud.githubusercontent.com/assets/10035308/8245309/a575cc8c-15e9-11e5-8735-0a4ab2a4e137.png)
## Neo Geo Pocket

![neogeopocketdiagram](https://cloud.githubusercontent.com/assets/10035308/8244887/0e06c54a-15e4-11e5-8f8f-28758d16c446.png)
## PS3

![playstation](https://cloud.githubusercontent.com/assets/10035308/8194532/8213433e-1439-11e5-8e69-87c6a7dc275c.png)
## SG-1000

![sg-1000diagram](https://cloud.githubusercontent.com/assets/10035308/8238988/4aee1770-15b6-11e5-90d4-5bf66ac7e3dd.png)
## Super Nintendo

![snes](https://cloud.githubusercontent.com/assets/10035308/7334403/bd655d6e-eb4e-11e4-8fd7-a4424aad1034.png)
## Sega Saturn

![saturn](https://cloud.githubusercontent.com/assets/10035308/7449063/327bfb76-f1e9-11e4-9c58-4b38a3c1284d.png)
## Turbografx16

![turbografx16diagram](https://cloud.githubusercontent.com/assets/10035308/8239221/aa5f3602-15b7-11e5-9c9f-54322c0a18bc.png)
## Videopac/Odyssey2

![videopacdiagram](https://cloud.githubusercontent.com/assets/10035308/8236975/679e6a58-15a9-11e5-98de-4ca681404eca.png)
## Vectrex

![vectrex](https://cloud.githubusercontent.com/assets/10035308/8196876/168671c6-144e-11e5-8139-d2d980a936fa.png)
## VirtualBoy

![virtualboydiagram](https://cloud.githubusercontent.com/assets/10035308/8246349/6552ef7a-15fa-11e5-9a78-1800b51cccc5.png)

---
## Intellivision

![intellivision](https://cloud.githubusercontent.com/assets/10035308/8246393/3e98c8c2-15fb-11e5-9398-3f5abd60361b.png)
## Sega Dreamcast

![dreamcast](https://cloud.githubusercontent.com/assets/10035308/8246417/9c839e08-15fb-11e5-9747-cf0b25727c11.png)
## Xbox 360

![xboxdiagram](https://cloud.githubusercontent.com/assets/10035308/8268487/b34244f6-1744-11e5-8ab0-58037d5e096a.png)
## PlayStation 3

![ps3](https://cloud.githubusercontent.com/assets/10035308/8268491/ecf1dedc-1744-11e5-9fae-00c7c8aaad06.png)

**Comment by taalas** [comment]
Those look great and fit very well in the Wiki pages imho! Great job!

**Comment by taalas** [comment]
I noticed that in some of the diagrams the color gradients for the controller bodies seem to be faulty. This might be due to resizing the controller and the gradient anchors not moving as well (I think this might be an Inkscape problem). It's very visible on the second and third but last schematic in the post above...

**Comment by HerbFargus** [comment]
Yeah I just realised that, my laptop screen has horrible quality so I didn't quite notice and yes I think you may be right it is an issue with inkscape. this seems to be a viable fix potentially:
http://inkscapetutorials.org/2011/02/15/inkscape-faq-why-do-my-gradients-change-when-i-move-or-resize-an-object/ 
I may try it out and see if I can redo those diagrams. (guess I should learn how to use the program a little better :p )

**Comment by HerbFargus** [comment]
Ok: These are the before and after. second one is the newest one- I'll change the others with the gradient option selected in inkscape.
![philips_videopac](https://cloud.githubusercontent.com/assets/10035308/8192731/79eeaea2-142d-11e5-924d-9d284d280981.png)
![videopacdiagram](https://cloud.githubusercontent.com/assets/10035308/8236975/679e6a58-15a9-11e5-98de-4ca681404eca.png)

**Comment by taalas** [comment]
Looks much better now, thanks for taking the time to look it over again! Not your fault...I think Inkscape's behaviour in regards to gradients is often times very dubious ^^ thanks for the link!

**Comment by HerbFargus** [comment]
Ok, so this is more of a question of personal preference. Thus far my method has been to start with a paper space of 1920X1080 and resize the controllers to all fit within that space as best as possible just for consistency's sake, and then I save all the SVGs with the same 1920X1080 paperspace but for the final png files that go onto this page and subsequently the retropie wiki I've resized it to page content so there isn't all this extra white space. Would you prefer I resize the svgs to page content as well rather than 1920X1080 paperspace?

**Comment by taalas** [comment]
Your current approach with having a fixed page size of 1920x1080 for the vector-based formats and then exporting with the page fit-to-content is totally fine with me!

Maybe I should have gone for a fixed page size for the controller sources as well. I do try to construct the controllers such as that the relative sizes between them somehow reflects the real proportions (I have them in a single huge file and separate them from there). But when saving the single files I fit the page to the content (as you do for the raster graphics exports).

If it's ok with you I'd propose we continue with the way the files are now.

**Comment by HerbFargus** [comment]
I figure as long as the proportions for the controllers are correct with relation to itself, that's most important as these are mostly going to be on their own pages rather than being with other controllers. So how you've done it is great :) I should be able to redo all the controllers and finish up the ones you just added by the end of the day.  Once everything is finished we can discuss the best way to add controllers to the [RetroPie wiki](https://github.com/RetroPie/RetroPie-Setup/wiki). I'd also like to get these on the [libretro wiki](http://wiki.libretro.com/index.php?title=Main_Page) if at all possible.

**Comment by taalas** [comment]
I will try to finish the rest of the controllers tomorrow I hope. I also thought about asking the Libretro project if they were interested in the graphics. I think it would make a great addition to their wiki as well...the tables they currently use are ok, having the controller as a base for button placement would make a nice addition though imho.

Lakka.tv does have some controller art though that they use in their CrossMediaBar-inspired menu...or alt least it seems from the screenshots/mock-ups I have seen. I haven't tried Lakka myself yet...

**Comment by HerbFargus** [comment]
Sounds good, I haven't checked out lakka either, I've seen videos on youtube of it, but never installed it myself. So a couple questions on retroarch controls- the documentation is kinda sparse (quite frankly this is probably the first real documentation on it) 

For Virtual Boy- is the right dpad used by the right analogue stick? (I don't have any controllers with an analogue to test)

Are Dpad controls interchangeable with Analogue sticks? this changelog seems to suggest that they might/could be: http://www.libretro.com/index.php/retroarch-1-0-0-2-changelog/

**Comment by taalas** [comment]
I am not completely sure regarding the Virtual Boy mappings, but I can test it on my system since I am using an Xbox 360 pad as my main control right now.

As for the Dpad controls and analogue sticks it always seemed to me that it would make a difference with different Retroarch cores. I have an open issue with RetroPie that includes a kind of similar "problem": when using my Xbox 360 controller with the wireless receiver a specific RetroArch autoconfig file is used that maps the Dpad as an analog input (and thus to a RetroPad *_axis mapping). At least for the cores I tested this seems not to work for me (no directional input possible). I have changed my driver configuration to use the Xbox 360 digital dpad as digital inputs and the shoulder buttons as analog inputs and use a modified autoconfig file for this. For me this does seem to represent the real controller much better than the default config that is currently used.

I also talked to the RetroArch people about this but couldn't get any definitive answer.

There is also another open issue from yesterday at RetroPie I think where someone seems to have the same problem with his wireless controller.

Not sure if I may have misunderstood your question though ;) It would theoretically be possible to map an analog stick to the (analog version) of the RetroPad Dpad mappings...in my experience not all cores can use this though...

**Comment by HerbFargus** [comment]
Yeah from what I've read it hasn't been very clear- the only one that really seems to work on switching from dpads to analog sticks is the PlayStation and that's only after the retroarch.cfg is changed to one or the other, rather than it being automatic. I was mostly just curious as all my diagrams have dpad inputs and just wanted to make sure I wasn't putting dpad when it should have been analog.

**Comment by taalas** [comment]
Nah, I think it's totally fine to map all those pads as digital pads!

As much as I like RetroArch and the idea behind it, their documentation in regards to controllers and controller configuration is very lacking imho. When debugging my pads I was hoping for some kind of overview over their variables regarding controller config files, but couldn't find anything except the joypad config program that produces those files (no real API reference). Perhaps we can improve that a bit with our efforts...

**Comment by HerbFargus** [comment]
Yeah documentation is lacking a lot- I think part of it is that libretro/retroarch are typically used in the backend and managed through things like emulationstation/lakka. but what we're doing will definitely help.

In regards to the Dpad question, I guess Floob answered my questions with his latest video: https://www.youtube.com/watch?v=mEg_OrlMUFU he shows how to use the analog with with the dpad as well- looks like its interchangeable for most if not all the retroarch emulators. good to know.

**Comment by taalas** [comment]
Ah, yes, that's good to know!

**Comment by HerbFargus** [comment]
Also in regards to controller configs it looks like they recently added a core remapping option in the rgui where you can change the default core inputs so that instead of the default buttons on the diagrams it would allow people to map the retropad controller differently. It doesn't work on the current build but it is there in the menu.

http://blog.petrockblock.com/forums/topic/core-input-remapping-options/

**Comment by taalas** [comment]
That is an amazing option, one I have hoped for for a while. For me, the standard mappings of almost any 2 button console to be A and B contradicts with the way I hold a (modern) gamepad. I'd much rather have the 2 buttons at the tip and the bottom of my thumb...

**Comment by HerbFargus** [comment]
I just tested the new build (retropie RC1) with the core input remapping options and it worked pretty well, unfortunately its only on some emulators so imame4all libretro still needs to be modified manually in order for the coin insert button to work, but I think fba-libretro has core input remapping.

**Comment by taalas** [comment]
Hi and sorry for the delay. My wedding took some time away from the project ;) I still haven't tested the core remapping options but it is good to know that it seems to work now. What I would still love to have is some kind of emulator mapping graphic once an emulator is started. I will detail that in another issue...

**Comment by HerbFargus** [comment]
well congratulations!-  your marriage is far more important- gizmo98 just got married as well, and I just had my one year anniversary :) I think a graphical interface is a brilliant idea- It would be cool as part of the retropie project but as far as coding goes I think it would be far more beneficial for it to be directly integrated into retroarch rather than specific to retropie (which I'm guessing you've probably already considered) I'm starting a programming course soon so maybe after i finish it I may be more useful in the coding department.
